File: backend/src/main/java/com/revnu/backend/features/expenses/dto/ExpenseResponse.java
package com.revnu.backend.features.expenses.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

import com.revnu.backend.features.expenses.model.ExpenseStatus;

public record ExpenseResponse(
        UUID id,
        BigDecimal amount,
        UUID categoryId,
        String categoryName,
        String notes,
        UUID fileId,
        ExpenseStatus status,
        LocalDateTime createdAt
        ) {

}

File: mobile/app/src/main/java/com/revnu/mobile/features/expenses/model/ExpenseResponse.kt
package com.revnu.mobile.features.expenses.model

import com.google.gson.annotations.SerializedName

data class ExpenseResponse(
    @SerializedName("id") val id: String,
    @SerializedName("amount") val amount: Double,
    @SerializedName("categoryId") val categoryId: String,
    @SerializedName("categoryName") val categoryName: String?,
    @SerializedName("notes") val notes: String?,
    @SerializedName("fileId") val fileId: String?,
    @SerializedName("status") val status: String,
    @SerializedName("createdAt") val createdAt: String
)